src/eduid_webapp/email/verifications.py

```diff
@@ -121,9 +122,16 @@ def verify_mail_address(state, proofing_user):
         if has_primary is None:
             proofing_user.mail_addresses.find(state.verification.email).is_primary = True
 
-    save_and_sync_user(proofing_user)
-    current_app.logger.info('Email address {!r} confirmed '
-                            'for user {}'.format(state.verification.email, proofing_user))
-    current_app.stats.count(name='email_verify_success', value=1)
-    current_app.proofing_statedb.remove_state(state)
-    current_app.logger.debug('Removed proofing state: {} '.format(state))
+    mail_address_proofing = MailAddressProofing(proofing_user, created_by='email', mail_address=new_email.email,
+                                                reference=state.reference, proofing_version='2013v1')
+    if current_app.proofing_log.save(mail_address_proofing):
```

Comment:
Please do

if not current_app.proofing_log.save(mail_address_proofing):
    current_app.logger.error('Could not save email proofing to proofing log')
    return False

to minimize the context I have to remember when reading the code.